app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -49,14 +50,13 @@ class PromotedStoryListViewModel(
     routeToRecentlyPlayedListener.routeToRecentlyPlayed()
   }
 
+  // Overriding equals is needed so that DataProvider combine functions used in the HomeViewModel
+  // will only rebind when the actual data in the data list changes, rather than when the ViewModel
+  // object changes.
   override fun equals(other: Any?): Boolean {
-    if (this === other) {
-      return true
-    }
-    if (other == null || other.javaClass != javaClass) {
-      return false
-    }
-    val otherResult = other as PromotedStoryListViewModel
-    return otherResult.promotedStoryList == this.promotedStoryList
+    return other is PromotedStoryListViewModel &&
+      other.promotedStoryList == this.promotedStoryList
   }
+
+  override fun hashCode() = Objects.hash(this.promotedStoryList)
```

Comment:
Nit: strike 'this' here & elsewhere where it isn't needed (we generally only use 'this' when it's actually needed for consistency).

Reply:
Done